lib/modules/datasource/docker/dockerhub-cache.ts

```diff
@@ -32,32 +32,64 @@ export class DockerHubCache {
     return new DockerHubCache(dockerRepository, repoCache);
   }
 
-  reconcile(items: DockerHubTag[]): boolean {
+  reconcile(items: DockerHubTag[], expectedCount: number): boolean {
     let needNextPage = true;
 
+    let earliestDate = null;
+
     let { updatedAt } = this.cache;
     let latestDate = updatedAt ? DateTime.fromISO(updatedAt) : null;
 
+    const ids = new Set<number>();
+
     for (const newItem of items) {
       const id = newItem.id;
+      ids.add(id);
+
       const oldItem = this.cache.items[id];
 
+      const itemDate = DateTime.fromISO(newItem.last_updated);
```

Comment:
Do we need validation here? Maybe DateTime.fromISO could fail?

Reply:
Doesn't happen in practice it seems, but maybe in separate PR we could do it just in case (on schema level)

Reply:
Are you validating the date using zod instead?

Reply:
No, we don't, so far it goes without validation

Reply:
In that case please wrap it in try/catch because I think it would throw if encountering an invalid string

Reply:
It doesn't throw, it returns special invalid `DateTime` instance

Reply:
Ah, yes, we did the string validation at the schema level